### PR TITLE
Cray libsci and ver

### PR DESCRIPTION
**Pull request by mamelara**
Not intending for this to be merged yet until reviewed and fully tested but this PR does a couple of things.

1. Appends a major version number to the CNL os version in addition to making CNL lowercase.
- Recently found out that the compute node linux os version is found by the version given by PrgEnv-X. This is how Cray displays their os version to NERSC, not sure if this is the same across all Crays.

2. unset CRAY_LD_LIBRARY_PATH
- Unless someone tells me this is a bad idea, this should unset CRAY_LD_LIBRARY_PATH.

3. Adds a dummy package for Cray-libsci (provides lapack, blas and scalapack)
- Used so that Spack is aware that a cray-libsci package exists.

Something I forgot to do is unload cray-libsci in the build-environment prior to building. This could potentially cause issues if a user on a Cray machine wishes to link with MKL or any other lapack, blas, scalapack provider due to the -L statement to cray-libsci found by default in the cray compiler wrappers.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cray-libsci/package.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#
+from spack import *
+
+
+class CrayLibsci(Package):
+    """The Cray Scientific Libraries package, LibSci, is a collection of
+    numerical routines optimized for best performance on Cray systems."""
+
+    homepage = "http://www.nersc.gov/users/software/programming-libraries/math-libraries/libsci/"
+    url      = "http://www.nersc.gov/users/software/programming-libraries/math-libraries/libsci/"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cray-libsci/package.py]: This isn't a real URL. I would just remove it.

Comment (mamelara) [inline var/spack/repos/builtin/packages/cray-libsci/package.py]: So, I removed it in the next commit but now I run into this error whenever I try `spack spec blas` or `spack spec cray-libsci`

```
==> Error: Class constructor failed for package 'cray-libsci'.

Caused by:
NoURLError: Package CrayLibsci has no version with a URL.
  File "/global/u2/m/mamelara/spack/lib/spack/spack/repository.py", line 580, in get
    self._instances[key] = package_class(copy)
  File "/global/u2/m/mamelara/spack/lib/spack/spack/package.py", line 562, in __init__
    f = fs.for_package_version(self, self.version)
  File "/global/u2/m/mamelara/spack/lib/spack/spack/fetch_strategy.py", line 878, in for_package_version
    url = pkg.url_for_version(version)
  File "/global/u2/m/mamelara/spack/lib/spack/spack/package.py", line 682, in url_for_version
    raise NoURLError(cls)
```

Am I missing something else here?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cray-libsci/package.py]: That's a known bug: https://github.com/LLNL/spack/pull/3042

I guess you'll have to leave it in for now until @tgamblin or someone can fix it.

**Comment by mamelara** [comment]
Closing since #7583 covers half of what this PR does. I will open a separate PR to add cray-libsci as a package.

**Pull request closed without merging**